ts/components/services/ContactPreferencesToggles/index.tsx

```diff
@@ -1,91 +1,178 @@
-import React, { useState } from "react";
+import * as pot from "italia-ts-commons/lib/pot";
+import React, { useEffect, useMemo, useState } from "react";
 import { connect } from "react-redux";
 import { fromNullable } from "fp-ts/lib/Option";
-
 import { GlobalState } from "../../../store/reducers/types";
 import I18n from "../../../i18n";
 import ItemSeparatorComponent from "../../ItemSeparatorComponent";
 import { NotificationChannelEnum } from "../../../../definitions/backend/NotificationChannel";
+import { Dispatch } from "../../../store/actions/types";
+import {
+  servicePreferenceSelector,
+  ServicePreferenceState
+} from "../../../store/reducers/entities/services/servicePreference";
+import { ServiceId } from "../../../../definitions/backend/ServiceId";
+import {
+  loadServicePreference,
+  upsertServicePreference
+} from "../../../store/actions/services/servicePreference";
+import { serviceIDCurrentSelector } from "../../../store/reducers/entities/services/currentService";
+import {
+  isServicePreferenceResponseSuccess,
+  ServicePreference
+} from "../../../types/services/ServicePreferenceResponse";
+import { isStrictSome } from "../../../utils/pot";
+import { showToast } from "../../../utils/showToast";
 import PreferenceToggleRow from "./PreferenceToggleRow";
 
 type Item = "email" | "push" | "inbox";
 
 type Props = {
   channels?: ReadonlyArray<NotificationChannelEnum>;
-} & ReturnType<typeof mapStateToProps>;
+} & ReturnType<typeof mapStateToProps> &
+  ReturnType<typeof mapDispatchToProps>;
+
+const hasChannel = (
+  channel: NotificationChannelEnum,
+  channels?: ReadonlyArray<NotificationChannelEnum>
+) =>
+  fromNullable(channels)
+    .map(anc => anc.indexOf(channel) !== -1)
+    .getOrElse(true);
```

Comment:
```suggestion
) => (channels ?? []).indexOf(channel) !== -1;
```
Now that we have the nullish coalescing operator, I find this code is short and clear.
btw if `channels` is not defined do we consider that `channel` present? (`.getOrElse(true)`)

Reply:
this was the original logic, if we don't have channels value we consider as all the channels are available

Reply:
In this scenario your implementation it's better than mine because it will become at the best a ternary operator